homeassistant/components/bsblan/config_flow.py

```diff
@@ -41,6 +44,68 @@ async def async_step_user(
         self.username = user_input.get(CONF_USERNAME)
         self.password = user_input.get(CONF_PASSWORD)
 
+        return await self._validate_and_create()
+
+    async def async_step_zeroconf(
+        self, discovery_info: ZeroconfServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle Zeroconf discovery."""
+
+        self.host = str(discovery_info.ip_address)
+        self.port = discovery_info.port or DEFAULT_PORT
+
+        # Try to get MAC from properties first (for future compatibility)
+        self.mac = discovery_info.properties.get("mac")
+
+        # If MAC not found in properties, try to parse from raw TXT records
+        if not self.mac and hasattr(discovery_info, "properties_raw"):
+            for key in discovery_info.properties_raw:
+                if key.startswith(b"mac="):
+                    self.mac = key[4:].decode("utf-8")
+                    break
+
+        # If still no MAC found, abort the discovery flow
+        if not self.mac:
+            return self.async_abort(reason="no_mac_address")
+
+        # Check if the device is already configured using MAC
+        await self.async_set_unique_id(format_mac(self.mac))
+        self._abort_if_unique_id_configured(
+            updates={
+                CONF_HOST: self.host,
+                CONF_PORT: self.port,
+            }
+        )
+
+        # Proceed to get credentials
+        self.context["title_placeholders"] = {"name": f"BSBLAN {self.host}"}
+        return await self.async_step_discovery_confirm()
+
+    async def async_step_discovery_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle getting credentials for discovered device."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="discovery_confirm",
+                data_schema=vol.Schema(
+                    {
+                        vol.Optional(CONF_PASSKEY): str,
+                        vol.Optional(CONF_USERNAME): str,
+                        vol.Optional(CONF_PASSWORD): str,
+                    }
+                ),
+                description_placeholders={"host": str(self.host)},
+            )
+
+        self.passkey = user_input.get(CONF_PASSKEY)
+        self.username = user_input.get(CONF_USERNAME)
+        self.password = user_input.get(CONF_PASSWORD)
+
+        return await self._validate_and_create()
```

Comment:
Downside of using this shared method. If i now discover a device and make a typo and then its possible to modify the host, which isn't what we want